checks/shell_download_validate.go

```diff
@@ -401,7 +400,63 @@ func isPipUnpinnedDownload(cmd []string) bool {
 		}
 	}
 
-	return isInstalled
+	return isInstall
+}
+
+func isPythonCommand(cmd []string) bool {
+	for _, pi := range pythonInterpreters {
+		if isBinaryName(pi, cmd[0]) {
+			return true
+		}
+	}
+	return false
+}
+
+func isPythonPipInstall(cmd []string) bool {
+	if !isPythonCommand(cmd) {
+		return false
+	}
+
+	isInstall := false
+	hasPip := false
+
+	for i := 1; i < len(cmd); i++ {
+		// Search for pip module.
+		if strings.EqualFold(cmd[i], "-m") &&
```

Comment:
Maybe make this `if {}` and `else if {}` without `continue`, so that, we check for `isInstall && hasPip` at the end of every for loop and break if `true`.

Reply:
function re-written and split in two. ptal